compose/snippets/src/main/java/com/example/compose/snippets/graphics/ShapesSnippets.kt

```diff
@@ -0,0 +1,329 @@
+            .drawWithCache {
+                val triangle = RoundedPolygon(
+                    numVertices = 3,
+                    radius = size.width / 2f,
+                    centerX = size.width / 2f,
+                    centerY = size.height / 2f,
+                    rounding = CornerRounding(
+                        size.width / 10f,
+                        smoothing = 0.1f
+                    )
+                )
+                val square = RoundedPolygon(
+                    numVertices = 4,
+                    radius = size.width / 2f,
+                    centerX = size.width / 2f,
+                    centerY = size.height / 2f
+                )
+
+                val morph = Morph(start = triangle, end = square)
+                // todo ensure proper caching here.
+                val morphPath = morph
+                    .toComposePath(progress = morphProgress.value)
+
+                onDrawBehind {
+                    drawPath(morphPath, color = Color.Black)
+                }
+            }
+            .fillMaxSize()
+    )
+    // [END android_compose_graphics_polygon_morph_animation]
+}
+/**
+ * Transforms the morph at a given progress into a [Path].
+ * It can optionally be scaled, using the origin (0,0) as pivot point.
+ */
+fun Morph.toComposePath(progress: Float, scale: Float = 1f, path: Path = Path()): Path {
+    var first = true
+    path.rewind()
+    forEachCubic(progress) { bezier ->
+        if (first) {
+            path.moveTo(bezier.anchor0X * scale, bezier.anchor0Y * scale)
+            first = false
+        }
+        path.cubicTo(
+            bezier.control0X * scale, bezier.control0Y * scale,
+            bezier.control1X * scale, bezier.control1Y * scale,
+            bezier.anchor1X * scale, bezier.anchor1Y * scale
+        )
+    }
+    path.close()
+    return path
+}
+
+/**
+ * Function used to create a Path from a list of Cubics.
+ */
+fun List<Cubic>.toPath(path: Path = Path()): Path {
+    path.rewind()
+    firstOrNull()?.let { first ->
+        path.moveTo(first.anchor0X, first.anchor0Y)
+    }
+    for (bezier in this) {
+        path.cubicTo(
+            bezier.control0X, bezier.control0Y,
+            bezier.control1X, bezier.control1Y,
+            bezier.anchor1X, bezier.anchor1Y
+        )
+    }
+    path.close()
+    return path
+}
+
+class MorphPolygonShape(
+    private val morph: Morph,
+    private val percentage: Float
+) : Shape {
+
+    override fun createOutline(
+        size: Size,
+        layoutDirection: LayoutDirection,
+        density: Density
+    ): Outline {
+        /*val matrixA = calculateMatrix(shapeA.bounds, size.width, size.height)
+        shapeA.transform(matrixA)
+        val matrixB = calculateMatrix(shapeB.bounds, size.width, size.height)
+        shapeB.transform(matrixB)
+        val morph = Morph(shapeA, shapeB)
+        morph.progress = percentage*/
+        // todo figure out how to scale and offset the shapes here
```

Comment:
@chethaase What would be best to change the scale / offset of the shape without having access to the shape or recreating it here?